services/classification/luis/client.go

```diff
@@ -0,0 +1,72 @@
+package luis
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/nyaruka/goflow/utils"
+	"github.com/nyaruka/goflow/utils/httpx"
+
+	"github.com/shopspring/decimal"
+)
+
+type ExtractedIntent struct {
+	Intent string          `json:"intent"`
+	Score  decimal.Decimal `json:"score"`
+}
+
+type ExtractedEntity struct {
+	Entity     string          `json:"entity"`
+	Type       string          `json:"type"`
+	StartIndex int             `json:"startIndex"`
+	EndIndex   int             `json:"endIndex"`
+	Score      decimal.Decimal `json:"score"`
+}
+
+type SentimentAnalysis struct {
+	Label string          `json:"label"`
+	Score decimal.Decimal `json:"score"`
+}
+
+// PredictResponse is the response from a predict request
+type PredictResponse struct {
+	Query             string            `json:"query"`
+	TopScoringIntent  *ExtractedIntent  `json:"topScoringIntent"`
+	Intents           []ExtractedIntent `json:"intents" validate:"required"`
+	Entities          []ExtractedEntity `json:"entities"`
+	SentimentAnalysis SentimentAnalysis `json:"sentimentAnalysis"`
+}
+
+// Client is a basic LUIS client
+type Client struct {
+	httpClient *http.Client
+	appURL     string
+	key        string
+}
+
+// NewClient creates a new client
+func NewClient(httpClient *http.Client, appURL, key string) *Client {
+	return &Client{
+		httpClient: httpClient,
+		appURL:     appURL,
```

Comment:
looks like LUIS endpoints can differ completely between apps so need full URL to be stored with classifier

Reply:
updated to take endpoint + appID + key